src/views/collections/CreateCollection/EditCollection.vue

```diff
@@ -0,0 +1,218 @@
+            <CollectionInfo
+              :title="title"
+              :body="desc"
+              :picture="picture"
+              @update:title="(text) => (title = text)"
+              @update:body="(text) => (desc = text)"
+              @update:image="(file) => (newPicture = file)"
+            />
+          </div>
+          <div class="col-md-6">
+            <CollectionPuzzles
+              :puzzlelist="puzzlelist"
+              @update:puzzle="(puzzles) => (puzzlelist = puzzles)"
+            />
+          </div>
+        </div>
+        <div class="row">
+          <div class="col-md-6">
+            <b-button type="submit" variant="primary" @click="submit()">
+              {{ $t('create-collection:collection-info:main-action') }}
+            </b-button>
+            <b-button
+              type="submit"
+              style="margin-left: 10px"
+              variant="danger"
+              @click="deleteCollection"
+            >
+              {{ 'Delete' }}
+            </b-button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </EternaPage>
+</template>
+
+<script lang="ts">
+// @ts-ignore
+  import debounce from 'lodash.debounce';
+  import {
+    Component,
+    Vue,
+    Mixins,
+    Watch,
+    Ref,
+    Prop,
+  } from 'vue-property-decorator';
+  import { RouteCallback, Route } from 'vue-router';
+  import axios, { AxiosInstance } from 'axios';
+  // @ts-ignore
+  import EternaPage from '@/components/PageLayout/EternaPage.vue';
+  import TagsPanel from '@/components/Sidebar/TagsPanel.vue';
+  import Utils from '@/utils/utils';
+  import { CollectionItem, PuzzleItem } from '@/types/common-types';
+  import CollectionInfo from './components/CollectionInfo.vue';
+  import CollectionPuzzles from './components/CollectionPuzzles.vue';
+
+  @Component({
+    components: {
+      EternaPage,
+      TagsPanel,
+      CollectionInfo,
+      CollectionPuzzles,
+    },
+  })
+  export default class EditCollection extends Vue {
+    @Prop({ required: true }) private loading!: boolean;
+
+    private title = '';
+
+    private desc = '';
+
+    private puzzlenames: PuzzleItem[] = [];
+
+    private puzzlelist: PuzzleItem[] = [];
+
+    private picture: string = '';
+
+    private newPicture: File | null = null;
+
+    private data: boolean = false;
+
+    fetchData: () => Promise<void> | undefined = async () => {};
+
+    mounted() {
+      this.fetch();
+    }
+
+    async fetch() {
+      const collection = (
+        await this.$http.get(`/get/?type=collection&nid=${this.$route.params.id}`)
+      ).data.data.collection as CollectionItem;
+      if (collection.puzzles != null) {
+        const puzzlelist = collection.puzzles.split(',');
+        Object.values(puzzlelist).forEach(async (puzz) =>
+          this.puzzlelist.push(
+            (await (
+              await this.$http.get(`/get/?type=puzzle&nid=${parseInt(puzz, 10)}`)
+            ).data.data.puzzle) as PuzzleItem,
+          ),
+        );
```

Comment:
Couple things here:
- Prefer `.map` to looping over a list and pushing derived values to a list
- It looks like you have an extra `await` tat you don't need
- I believe what you're doing here is making fetch calls in serial instead of in parallel. You should construct an array of promises (ie, the thing that gets returned when you call an async function without awaiting it) and then call `Promise.all(arrayOfPromises)` and then await _that_ so that the calls get parallelized
- That being said, if you need the data from all the puzzles, maybe it would be better to do that in the backend instead and return it along with the collection data? That way you could also fetch all of them in one DB query and just serialize as an array, vs incurring overhead both from all the network calls and individual DB queries.